packages/renderer/src/lib/onboarding/telemetry.ts

```diff
@@ -0,0 +1,134 @@
+export interface TelemetryStep {
+  id: string;
+  title: string;
+  durationMs: number;
+  error: undefined | string;
+}
+
+export interface TelemetryData {
+  extension: string;
+  skipped: boolean;
+  steps: TelemetryStep[];
+  stepsDetails?: any;
+  durationMs: number;
+  skipAtStep?: string;
+  errors?: string[];
+}
+
+export class TelemetrySession {
```

Comment:
just a minor thing but as it's scoped only for onboarding, maybe we should prefix it with Onboarding so we're not tempted to use it elsewhere than the onboarding.

TelemetrySession --> OnboardingTelemetrySession